packages/engine/src/store/types.ts

```diff
@@ -135,10 +135,19 @@ export type SketchModules = { [key: string]: SketchModuleItem }
 
 export type EnumOption = { value: string; label: string }
 
+export interface Input {
+  id: string
+  type: 'midi' | 'gamepad' | string
```

Comment:
haven't really thought type through, might want to use the event types from the midi package, unused atm, could delete

Reply:
It's fine for now, I think this is just going to naturally evolve as we add more inputs and realise what we do and don't need! I think for now though you can just keep it as `midi` with no extras and we just add to that.